services/CurrentTime/include/ble-service-current-time/CurrentTimeService.h

```diff
@@ -0,0 +1,178 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef BLE_CURRENT_TIME_SERVICE_H
+#define BLE_CURRENT_TIME_SERVICE_H
+
+#ifdef BLE_FEATURE_GATT_SERVER
+
+#include "ble/BLE.h"
+#include "ble/Gap.h"
+#include "events/EventQueue.h"
+#include "mbed_rtc_time.h"
+
+#include <time.h>
+
+/**
+ * BLE Current Time Service
+ *
+ * @par purpose
+ *
+ * @par usage
+ *
+ * @note The specification for the current time service can be found here:
+ * https://www.bluetooth.com/specifications/gatt
+ *
+ * @attention The user should not instantiate more than a single current time service service
+ */
+class CurrentTimeService : private ble::Gap::EventHandler {
+public:
+    struct CurrentTime {
+        uint16_t year;
+        uint8_t  month;
+        uint8_t  day;
+        uint8_t  hours;
+        uint8_t  minutes;
+        uint8_t  seconds;
+
+        uint8_t  weekday;
+
+        uint8_t  fractions256;
+
+        uint8_t  adjustReason;
+    };
+
+    struct EventHandler {
+        /**
+         * On current time changed
+         *
+         * This function is called if the current time characteristic is changed by the client
+         *
+         * @attention This is an abstract function and should be overridden by the user.
```

Comment:
this function is not abstract, it has an no op implementation

Reply:
@paul-szczepanek-arm We provided default implementations for every handler so far. Maybe we should stay consistent. If new handlers are added, it won't require to change `= 0` into `{}`. I don't have strong opinion on this.

Reply:
I wasn't suggesting we make this abstract. I think default implementation is good because saves the user from having to define it if he only needs one event (OK, this one so far has only one event so this point is moot). I was just pointing out the comment contradicts the code.

Reply:
Thanks Paul. The comment is a misuse of the term "abstract function". This mistake is also present in LLS, so I'll correct it there too.